File: lib/src/main/java/platinum/twod/Sprite2D.java
package platinum.twod;

import platinum.ecs.Component;
import platinum.ecs.System;

import java.awt.*;
import java.awt.image.BufferedImage;

public class Sprite2D extends Component<RenderSystem2D> {
    BufferedImage img;

    public Sprite2D(BufferedImage img) {
        this.img = img;
    }

    @Override
    public void init(System system) {
    }

    @Override
    public void update(System system) {
        Transform2D transform = getComponent(Transform2D.class);
        ((RenderSystem2D)system).graphics2D.drawImage(img, transform.getActXi(), transform.getActYi(), img.getWidth(), img.getHeight(), new Color(0, 0, 0, 0), null);
    }
}

File: lib/src/main/java/platinum/twod/Sprite2D.kt
package platinum.twod

import platinum.ecs.Component
import platinum.ecs.System
import java.awt.Color
import java.awt.image.BufferedImage

class Sprite2D(var image: BufferedImage) : Component<RenderSystem2D?>() {

    override fun init(system: System?) {}
    override fun update(system: System) {
        val transform = getComponent<Transform2D>() ?: return
        (system as RenderSystem2D).graphics2D!!.drawImage(image, transform.actXi, transform.actYi, image.width, image.height, Color(0, 0, 0, 0), null)
    }
}